test/rest_api/test_storage_service.py

```diff
@@ -470,3 +470,12 @@ def test_storage_service_keyspace_upgrade_sstables(cql, this_dc, rest_api):
                 # non-existing table
                 resp = rest_api.send("GET", f"storage_service/keyspace_upgrade_sstables/{keyspace}", { "cf": f"{test_tables[0]},XXX" })
                 assert resp.status_code == requests.codes.bad_request
+
+def test_storage_service_system_keyspace_repair(rest_api):
+    resp = rest_api.send("POST", f"storage_service/repair_async/system")
+    resp.raise_for_status()
+    assert resp.json() == 1, "Repair got invalid sequence number"
+
+    resp = rest_api.send("GET", "task_manager/list_module_tasks/repair")
+    resp.raise_for_status()
+    assert not resp.json(), "Repair of system keyspace didn't finish immediately"
```

Comment:
we should wait for the task to handle the race.
As for verifying the "immediately" part, we can measure wall clock and make sure it's fast enough, although I do hate the idea...

Reply:
For local keyspace's repairs new repair_id is created, but corresponding task is not. So maybe it should be "Repair task for keyspace with local replication strategy was created"?

We could get false positives though (if 10s is not enugh). If we want to avoid that, then we can set ttl to some large number. That will unfortunately make us move the test to task_manager_test.py to skip the test in release mode. I think it is not where it belongs to, though.

The reason why it keeps failing is that it is run multiple times on the same scylla instance, each time getting the next sequence number. Changing line 477 to `assert resp.json() > 0, "Repair got invalid sequence number"` will solve that.